src/pages/en/migrate.md

```diff
@@ -34,6 +34,65 @@ Astro no longer supports components or JSX expressions in Markdown pages by defa
 
 To make migrating easier, a new [legacy flag](/en/reference/configuration-reference/#legacyastroflavoredmarkdown) can be used to re-enable previous Markdown features.
 
+### Converting existing `.md` files to `.mdx`
+
+If you're not familiar with MDX, here are some steps you can follow to quickly convert an existing "Astro Flavored Markdown" file to MDX. As you learn more about MDX, feel free to explore other ways of writing your pages!
+
+1. Install the [`@astrojs/mdx`](/en/guides/integrations-guide/mdx/) integration.
+
+1. Change your existing `.md` file extensions to `.mdx`
+
+1. Remove any `layout:` and `setup:` properties from your frontmatter, replacing them with ESM import statements below the frontmatter.
+
+1. Wrap your MDX content in a `<Layout>` component, and pass all your frontmatter values as props.
+
+```mdx
+// src/pages/posts/my-post.mdx
+---
+title: md to mdx
+date: 2022-07-26
+tags: ["markdown", "mdx", "astro"]
+---
+import ReactCounter from '../../components/ReactCounter.jsx'
+import BaseLayout from '../../layouts/BaseLayout.astro'
+
+<BaseLayout { ...frontmatter}>
+    # {frontmatter.title}
+    
+    Here is my counter component, working in MDX:
+    
+    <ReactCounter client:load />
+</BaseLayout>
+```
+
+5. Update your layout component to no longer receive the Markdown content object, and update your template accordingly:
+
+```astro
+---
+// src/layouts/BaseLayout.astro
+
+const { title, date, tags } = Astro.props
+---
+
+- <h1>{content.title}</h1>
++ <h1>{title}</h1>
+
+<slot />
+
+```
+
+6. Update any `Astro.glob()` statement that currently retrieve `.md` files so that they will now return your `.mdx` files.
+
+:::tip
+While you are transitioning to MDX, you may wish to include both **`.md` and `.mdx`** files, so that your site continues to function normally even before all your files have been converted. Here is one way you can do that:
```

Comment:
```suggestion
While you are transitioning to MDX, you may wish to [enable the legacy flag](/en/reference/configuration-reference/#legacyastroflavoredmarkdown) and include both **`.md` and `.mdx`** files, so that your site continues to function normally even before all your files have been converted. Here is one way you can do that:
```